Feature/1 db model creation

B1iz0 opened the pull request.

B1iz0 committed:
Created initial entities of ReservationSystem Db

Added cs files with initial entities: Airplane, AirplaneType, Airport, City, Country, Flight, Role, User.
Annotations were added for each entity for some fields. Entities can be expanded in the future.

B1iz0 committed:
Created initial context for ReservationSystem Db

Context for our db was added and filled with DbSets of entities. There are: Roles, Users, AirplaneTypes, Airplanes, Flights, Countries, Cities, Airports.

B1iz0 committed:
Created an interface IRepository

This base repository contains prototypes of CRUD operations and will be used by future Repositories.

B1iz0 committed:
Created Booking entity for ReservationSystem Db

Booking entity with info about User and Flight was added. Also it required a change in User and Flight entities
List<Booking> was added to this entities.

B1iz0 committed:
-Created Repository for Airplane

-Created Repository for AirplaneType

B1iz0 committed:
-Renamed classes of entities (Example: Airplane -> AirplaneEntity)

-Deleted useless interface IUnitOfWork

B1iz0 committed:
Restructured of Asp Web Api

-Folder src for full project
-Renamed some projects (Current: ReservationSystemApi, Business, Data)

B1iz0 committed:
-Added initial migration

-Deleted annotations from classes of entities

B1iz0 committed:
-Added logging to console and file (Serilog lib)

B1iz0 committed:
-New migration (new table 'Companies')Companies

-Deleted useless using in entities classes

B1iz0 committed:
-FluentAPI was used to configure tables and properties

B1iz0 committed:
-Added repositories to work with data in db

B1iz0 committed:
-Deleted folder 'DataContext'

-Deleted useless using in repositories

B1iz0 committed:
-Added PasswordHasher class to hash user password

-Added initial data for Roles and Users tables (InitializingWithDefaultRoles migration)

B1iz0 committed:
-Changed initial info about admin user

B1iz0 committed:
-Added service for tokens

DzmitryAliakseichyk commented on src/Data/Entities/AirplaneTypeEntity.cs:
```diff
@@ -0,0 +1,11 @@
+    {
+        public int Id { get; set; }
+        public string TypeName { get; set; }
+        public List<AirplaneEntity> Airplanes { get; set; } = new List<AirplaneEntity>();
```
Auto init should be removed from model

DzmitryAliakseichyk commented on src/Data/Entities/AirportEntity.cs:
```diff
@@ -0,0 +1,16 @@
+        public CityEntity City { get; set; }
+        public int CompanyId { get; set; }
+        public CompanyEntity Company { get; set; }
+        public List<FlightEntity> ArrivingAirplanes { get; set; } = new List<FlightEntity>();
```
remove auto init

DzmitryAliakseichyk commented on src/Data/Entities/CityEntity.cs:
```diff
@@ -0,0 +1,13 @@
+        public string Name { get; set; }
+        public int CountryId { get; set; }
+        public CountryEntity Country { get; set; }
+        public List<AirportEntity> Airports { get; set; } = new List<AirportEntity>();
```
remove = new

DzmitryAliakseichyk commented on src/Data/PasswordHasher.cs:
```diff
@@ -0,0 +1,37 @@
+            return Convert.ToBase64String(salt);
+        }
+
+        //public static bool CheckHash(string password, string hash)
```
remove commented code

DzmitryAliakseichyk commented on src/Data/Repositories/AirplaneRepository.cs:
```diff
@@ -0,0 +1,52 @@
+        }
+        public IEnumerable<AirplaneEntity> GetAll()
+        {
+            return db.Airplanes;
```
extract DBSet for each repo in separate readonly field

DzmitryAliakseichyk reviewed (changes requested):
- remove = new List<Entity>(); everywhere
 - extract comment generic repo
 - u miss DbContext.SaveChanges() during CRUD

B1iz0 committed:
-Removed auto init of lists from entities

-Extracted DbSet for each repository readonly field
-Added SaveChanges() during CRUD

B1iz0 committed:
-Added Jwt token authorization with test data

B1iz0 committed:
-Added UsersController with CRUD operations

-Deleted useless section in appsettings
-Extended AppSettings class
P.s. I use UserEntity in GetAll method in Controller for now, but will use mapping a bit later

B1iz0 committed:
-Added virtual mod. for navigation properties of Entities for LazyLoading

B1iz0 committed:
-Added mapper for User,UserEntity

B1iz0 committed:
-Deleted useless attribute class

B1iz0 committed:
-Added claim role for tokens

-Added policy role "AdminApp"

B1iz0 committed:
-Added base mapper for mapping entities into models and vice versa

-Added some dto classes in Api and Business layers(user models and authenticate)
-Added Mappers(for user and auth models)

B1iz0 committed:
-Replaced base mapper with only user mapper

B1iz0 committed:
-Right using of user mapper
